tests/integration/utils.py

```diff
@@ -55,14 +43,86 @@ def create_user(
         session.add(user_obj)
         session.flush()
 
-        # return user updated with uuid:
+        # add token:
+        now = datetime.now()
+        future = now + timedelta(hours=100)
+        token = security.generate_token()
+        token_obj = models.ApiKey(
+            user_id = user_obj.id,
+            value = security.hash_token(token),
+            label = "valid_token",
+            expires = future
+        )
+        session.add(token_obj)
+        session.flush()
+
+        # add expired token:
+        now = datetime.now()
+        past = now - timedelta(hours=100)
+        expired_token = security.generate_token()
+        expired_token_obj = models.ApiKey(
+            user_id = user_obj.id,
+            value = security.hash_token(expired_token),
+            label = "expired_token",
+            expires = past
+        )
+        session.add(expired_token_obj)
+        session.flush()
+
+
+        # return user updated with uuid and tokens:
         user_updated = deepcopy(user)
```

Comment:
is there a reason why user is copied and not directly operated on?

Reply:
Yes because any modification to the user input inside the function would also affect the original object outside the function.

E.g.
```
def func(test):
    test["bla"] = "blup"

test_original = {"bla": "bla"}

func(test_original)
print(test_original)
#> {"bla": "blup"}
```